inference/core/interfaces/http/http_api.py

```diff
@@ -446,6 +451,70 @@ async def count_errors(request: Request, call_next):
                     self.model_manager.num_errors += 1
                 return response
 
+        if DEDICATED_DEPLOYMENT_WORKSPACE_URL:
+            cached_api_keys = set()
+            cached_projects = set()
+
+            @app.middleware("http")
+            async def check_authorization(request: Request, call_next):
+                # exclude / and /info (health check)
+                if request.url.path in ["/", "/info"]:
+                    return await call_next(request)
+
+                def _unauthorized_response(msg):
+                    return JSONResponse(
+                        status_code=401,
+                        content={
+                            "status": 401,
+                            "message": msg,
+                        },
+                    )
+
+                # check api_key
+                req_params = request.query_params
+                json_params = (
+                    await request.json()
+                    if request.headers.get("content-type", None) == "application/json"
+                    else dict()
+                )
+                api_key = req_params.get("api_key", None) or json_params.get(
+                    "api_key", None
+                )
+
+                if api_key not in cached_api_keys:
+                    try:
+                        workspace_url = (
+                            get_roboflow_workspace(api_key)
```

Comment:
If someone has the wrong API key (or say, a DDoS with a wrong API key), and they send many requests, this call to api.roboflow.com can result in rate-limiting, so maybe we should have a limit on how many such checks can be performed per minute by this server.

Reply:
Agree responsibility for enforcement of the rate limit is on the backend but the problem would be that an external malicious actor could get this client's IP blocked by coercing it into getting rate limited by the backend.